WeatherApp_Obj-C/WeatherApp_Obj-C/Screens/HomeScreen/HomeScreenViewController.m

```diff
@@ -20,17 +32,109 @@ - (void)viewDidLoad {
     [self showLoading];
     [[UserRepository new] getCurrentUser:^(FIRUser * _Nullable user) {
         [self hideLoading];
-        NSLog(@"%@", user.email);
     } error:^(NSString * _Nullable error) {
         [self hideLoading];
         [self showError:error];
     }];
     self.locationButton.backgroundColor = [UIColor greenButton];
-    self.locationButton.layer.cornerRadius = 20;
+    self.locationButton.layer.cornerRadius = smallButtonCornerRadius;
+    [self setupTableView];
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder
+{
+    self = [super initWithCoder:coder];
+    if (self) {
+        sunny = 0;
+        cloudy = 0;
+        raining = 0;
+        items = [[NSMutableArray alloc] init];
+    }
+    return self;
+}
+
+-(void)fetchData {
+    items = [[WeatherRepository new] fetchCDData];
+    [self.tableView reloadData];
 }
 
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:animated];
     [self.navigationController setNavigationBarHidden:YES animated:YES];
+    [self fetchData];
+    [self setupWeatherCounter];
 }
+
+-(void)setupTableView {
+    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
+    [self.tableView registerNib:[UINib nibWithNibName:NSStringFromClass([HomeScreenCell class]) bundle:nil] forCellReuseIdentifier:NSStringFromClass([HomeScreenCell class])];
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+}
+
+-(void)setupWeatherCounter {
+    [self resetWeatherCounter];
+    for (int i = 0; i < items.count; i++) {
+        NSString *main = items[i].mainEvent;
+        if ([main containsString:@"rain"] || [main containsString:@"thunderstorm"] || [main containsString:@"drizzle"]) {
+            raining++;
+        } else if ([main containsString:@"clouds"]) {
+            cloudy++;
+        } else if ([main containsString:@"clear"]) {
+            sunny++;
+        }
+        self.sunnyCounter.text = [NSString stringWithFormat:@"%d", sunny];
```

Comment:
Set text only after for loop, no point in setting it every time

Reply:
Solved in the next PR (Profile PR)